src/Components/CreateImageWizard/steps/Locale/components/KeyboardDropDown.tsx

```diff
@@ -0,0 +1,184 @@
+import React, { useEffect, useState } from 'react';
+
+import {
+  Button,
+  FormGroup,
+  MenuToggle,
+  MenuToggleElement,
+  Select,
+  SelectList,
+  SelectOption,
+  TextInputGroup,
+  TextInputGroupMain,
+  TextInputGroupUtilities,
+} from '@patternfly/react-core';
+import { TimesIcon } from '@patternfly/react-icons';
+
+import { useAppDispatch, useAppSelector } from '../../../../../store/hooks';
+import {
+  changeKeyboard,
+  selectKeyboard,
+} from '../../../../../store/wizardSlice';
+import { keyboardsList } from '../keyboardsList';
+
+const KeyboardDropDown = () => {
+  const keyboard = useAppSelector(selectKeyboard);
+  const dispatch = useAppDispatch();
+
+  const [isOpen, setIsOpen] = useState(false);
+  const [inputValue, setInputValue] = useState<string>('');
+  const [filterValue, setFilterValue] = useState<string>('');
+  const [selectOptions, setSelectOptions] = useState<string[]>(keyboardsList);
+
+  useEffect(() => {
+    let filteredKeyboards = keyboardsList;
+
+    if (filterValue) {
+      filteredKeyboards = keyboardsList.filter((keyboard: string) =>
+        String(keyboard).toLowerCase().includes(filterValue.toLowerCase())
+      );
+      if (!filteredKeyboards.length) {
+        filteredKeyboards = [`No results found for "${filterValue}"`];
+      }
+      if (!isOpen) {
+        setIsOpen(true);
+      }
+    }
+    setSelectOptions(filteredKeyboards.sort((a, b) => sortfn(a, b)));
+
+    // This useEffect hook should run *only* on when the filter value changes.
+    // eslint's exhaustive-deps rule does not support this use.
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [filterValue]);
+
+  const sortfn = (a: string, b: string) => {
```

Comment:
This is the same behavior as the package sort function as far as I can tell. What if we extracted this into a utility function? It would need to take a 3rd argument, the search term, but that would help DRY out the code.

Reply:
I added a commit for this little refactor so we don't have to go through a full edit/review cycle on Monday.